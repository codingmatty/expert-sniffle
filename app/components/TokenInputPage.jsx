import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';

export default function TokenInputPage() {
  const token = useSelector(({ token }) => token);
  const dispatch = useDispatch();
  const [tokenInput, setTokenInput] = useState(token);

  const onSubmit = (e) => {
    e.preventDefault();
    dispatch({ type: 'SET_GITHUB_TOKEN', token: tokenInput });
  };

  return (
    <div>
      <form onSubmit={onSubmit}>
        <input
          value={tokenInput}
          onChange={({ target: { value } }) => setTokenInput(value)}
        />
        <button type="submit">Save</button>
      </form>
      <div>Token: {token}</div>
    </div>
  );
}
