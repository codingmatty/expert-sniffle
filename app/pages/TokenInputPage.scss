.token-input-page {
  margin-top: 20%;
  max-width: 630px;

  p {
    text-align: justify;
  }

  .token-form {
    max-width: 500px;
    margin-top: 50px;
  }
}
