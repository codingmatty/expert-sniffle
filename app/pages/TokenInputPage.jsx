import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { navigate } from '@reach/router';

import { setGithubToken } from 'state/actions';

import Icon from 'components/Icon';

import './TokenInputPage.scss';

export default function TokenInputPage() {
  const token = useSelector(({ token }) => token);
  const dispatch = useDispatch();
  const [tokenInput, setTokenInput] = useState(token);

  const onSubmit = (e) => {
    e.preventDefault();
    dispatch(setGithubToken(tokenInput));
    navigate('/repos');
  };

  return (
    <div className="token-input-page mx-4 medium:mx-auto">
      <div>
        <p className="my-4 text-xlarge">Welcome!</p>
        <p className="my-4">
          Please provide a Github Personal Access Token to continue.
        </p>
        <p className="my-4">
          To obtain an Access Token, login to Github, visit the{' '}
          <a href="https://github.com/settings/tokens" target="_blank">
            Github Tokens <Icon name="open_in_new" />
          </a>
          , and generate a new one, providing a description like "Issue
          Prioritization Project" to describe its use and selecting the{' '}
          <code className="inline">repo</code> scope.
        </p>
        <p className="my-4">
          Once you enter it here, you will then be able to select a Github repo
          to show a list of Issues and prioritize them as needed.
        </p>
      </div>
      <form
        onSubmit={onSubmit}
        className="token-form mx-auto flex direction-column"
      >
        <label html="token-input" className="bold">
          Access Token:
        </label>
        <input
          id="token-input"
          className="mt-4"
          value={tokenInput}
          onChange={({ target: { value } }) => setTokenInput(value)}
          data-testid="token-input"
        />
        <button type="submit" className="mt-4">
          Submit
        </button>
      </form>
    </div>
  );
}
