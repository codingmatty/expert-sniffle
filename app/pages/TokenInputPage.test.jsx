import '@testing-library/jest-dom';
import configureStore from 'redux-mock-store';
import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import TokenInputPage from './TokenInputPage';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';

const mockStore = configureStore([thunk]);

describe('TokenInputPage', () => {
  test('sets redux state', () => {
    const store = mockStore({ token: '' })
    const { asFragment } = render(
      <Provider store={store}>
        <TokenInputPage />
      </Provider>
    );
    fireEvent.change(screen.getByTestId('token-input'), {
      target: { value: 'abcdef123456' },
    });
    fireEvent.click(screen.getByText(/submit/i));

    expect(store.getActions()[0]).toEqual(
      {
        type: 'SET_GITHUB_TOKEN',
        token: 'abcdef123456',
      }
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
