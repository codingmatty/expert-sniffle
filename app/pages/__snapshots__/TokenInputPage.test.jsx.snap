// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TokenInputPage sets redux state 1`] = `
<DocumentFragment>
  <div
    class="token-input-page mx-4 medium:mx-auto"
  >
    <div>
      <p
        class="my-4 text-xlarge"
      >
        Welcome!
      </p>
      <p
        class="my-4"
      >
        Please provide a Github Personal Access Token to continue.
      </p>
      <p
        class="my-4"
      >
        To obtain an Access Token, login to Github, visit the 
        <a
          href="https://github.com/settings/tokens"
          target="_blank"
        >
          Github Tokens 
          <i
            class="material-icons "
          >
            open_in_new
          </i>
        </a>
        , and generate a new one, providing a description like "Issue Prioritization Project" to describe its use and selecting the 
        <code
          class="inline"
        >
          repo
        </code>
         scope.
      </p>
      <p
        class="my-4"
      >
        Once you enter it here, you will then be able to select a Github repo to show a list of Issues and prioritize them as needed.
      </p>
    </div>
    <form
      class="token-form mx-auto flex direction-column"
    >
      <label
        class="bold"
        html="token-input"
      >
        Access Token:
      </label>
      <input
        class="mt-4"
        data-testid="token-input"
        id="token-input"
        value="abcdef123456"
      />
      <button
        class="mt-4"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
